File: dodam-core/src/main/java/b1nd/dodamcore/banner/domain/exception/BannerExceptionCode.java
package b1nd.dodamcore.banner.domain.exception;

import b1nd.dodamcore.common.exception.ExceptionCode;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

@RequiredArgsConstructor
public enum BannerExceptionCode implements ExceptionCode {

    NOT_FOUND(HttpStatus.NOT_FOUND, "없는 배너");

    private final HttpStatus status;
    private final String message;

    @Override
    public HttpStatus getHttpStatus() {
        return this.status;
    }

    @Override
    public String getExceptionName() {
        return this.name();
    }

    @Override
    public String getMessage() {
        return this.message;
    }

}

File: services/service-banner/src/main/kotlin/com/b1nd/dodamdodam/banner/domain/banner/exception/BannerExceptionCode.kt
package com.b1nd.dodamdodam.banner.domain.banner.exception

import com.b1nd.dodamdodam.core.common.exception.ExceptionCode
import org.springframework.http.HttpStatus

enum class BannerExceptionCode(
    override val status: HttpStatus,
    override val message: String,
) : ExceptionCode {
    BANNER_NOT_FOUND(HttpStatus.NOT_FOUND, "배너를 찾을 수 없습니다."),
}
